ostd/src/sync/mutex.rs

```diff
@@ -100,6 +101,16 @@ pub struct MutexGuard_<T: ?Sized, R: Deref<Target = Mutex<T>>> {
 /// A guard that provides exclusive access to the data protected by a [`Mutex`].
 pub type MutexGuard<'a, T> = MutexGuard_<T, &'a Mutex<T>>;
 
+impl<'a, T: ?Sized> MutexGuard<'a, T> {
+    /// # Safety
+    ///
+    /// The caller must ensure that the `MutexGuard`'s lifetime is properly managed. When the
+    /// `MutexGuard` goes out of scope, it will call `drop()` to release the mutex lock.
+    unsafe fn new(mutex: &'a Mutex<T>) -> MutexGuard<'a, T> {
```

Comment:
The safety precondition of this function should be that the passed `mutex` has successfully acquired the lock (as it's inner state). So it should be documented as:

```rust
    /// # Safety
    ///
    /// The caller must ensure that the given reference of [`Mutex`] lock has been successfully acquired in the
    /// current context. When the created [`MutexGuard`] is dropped, it will unlock the [`Mutex`].
```